File: util/EREnt.java
package common.util;

import common.io.json.JsonClass;
import common.io.json.JsonClass.NoTag;
import common.system.Copable;

import org.jetbrains.annotations.Nullable;

@JsonClass(noTag = NoTag.LOAD)
public class EREnt<X> implements BattleStatic, Copable<EREnt<X>> {

	@Nullable
	public X ent;
	public int multi = 100;
	public int mula = 100;
	public int share = 1;

	@Override
	public EREnt<X> copy() {
		EREnt<X> ans = new EREnt<X>();
		ans.ent = ent;
		ans.multi = multi;
		ans.mula = mula;
		ans.share = share;
		return ans;
	}

}

File: util/EREnt.kt
package common.util

import common.system.Copable

class EREnt<X> : BattleStatic, Copable<EREnt<X>?> {
    var ent: X? = null
    var multi = 100
    var mula = 100
    var share = 1
    override fun copy(): EREnt<X?> {
        val ans = EREnt<X?>()
        ans.ent = ent
        ans.multi = multi
        ans.mula = mula
        ans.share = share
        return ans
    }
}
